vault/login_mfa.go

```diff
@@ -270,6 +271,11 @@ func (i *IdentityStore) handleMFAMethodUpdateCommon(ctx context.Context, req *lo
 
 	methodID := d.Get("method_id").(string)
 
+	methodName, err := i.mfaBackend.sanitizeMethodNameSameNamespace(ns, d.Get("method_name").(string))
```

Comment:
Why do we want to assume that if the namespace path appears as a prefix of the method name, that it should be elided?  This isn't part of the api, is it?  I worry that it could lead to unintended behaviour.

Reply:
Sounds good! Thanks for the clear explanation. Basically, we want to allow whatever name an operator chooses for an MFA method. Then, in the validation, we make sure the method name has the namespace path prefixed.